src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -257,6 +258,97 @@ internal string GetString(int index, JsonTokenType expectedType)
             return lastString;
         }
 
+        internal bool TextEquals(int index, ReadOnlySpan<char> otherText, bool isPropertyName)
+        {
+            CheckNotDisposed();
+
+            int matchIndex = isPropertyName ? index - DbRow.Size : index;
+
+            (int lastIdx, string lastString) = _lastIndexAndString;
+        
+            if (lastIdx == matchIndex)
+            {
+                return otherText.SequenceEqual(lastString);
+            }
+
+            Span<byte> otherUtf8Text;
+            byte[] otherUtf8TextArray = null;
+
+            int length = otherText.Length;
+            if (length > JsonConstants.StackallocThreshold)
+            {
+                otherUtf8TextArray = ArrayPool<byte>.Shared.Rent(length);
+                otherUtf8Text = otherUtf8TextArray;
+            }
+            else
+            {
+                unsafe
```

Comment:
I don't think you need the unsafe code path here and you can probably follow a pattern like this:
https://github.com/dotnet/corefx/blob/40983e222a2fe12156360550275f89155a3cfc9c/src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.String.cs#L420-L422

So something like this:
```C#
byte[] otherUtf8TextArray = null;

int length = otherText.Length;
Span<byte> otherUtf8Text = length <= JsonConstants.StackallocThreshold ?
    stackalloc byte[length] :
    (otherUtf8TextArray = ArrayPool<byte>.Shared.Rent(length));
```